src/audio/host.c

```diff
@@ -489,15 +490,31 @@ static int host_params(struct comp_dev *dev)
 		return -EINVAL;
 	}
 
+#if defined CONFIG_DMA_GW
 	/* calculate period size based on config */
+	hd->period_bytes = ceil_divide(dev->frames * comp_frame_bytes(dev),
+				       PLATFORM_HDA_BUFFER_ALIGNMENT) *
+		PLATFORM_HDA_BUFFER_ALIGNMENT;
+#endif
+#if !defined CONFIG_DMA_GW
 	hd->period_bytes = dev->frames * comp_frame_bytes(dev);
+#endif
```

Comment:
This is rounding up period size and not buffer size. It could also have impact on scheduling since we are processing one period per schedule time slice. 
Also the DW DMA has optimal burst size and alignment too like the DMA GW, so perhaps we need an additional ops to be added to the DMA driver to do this operation and simplify host.c

Reply:
yes i was thinking a way to ask the dma alignment from the dma, but I still lack some knowledge about different dma's we have and their function. But yes exactly I would like to have method to get the alignment.

I guess the idea with the period_bytes alignment was, that if dma transfer _needs_ to be aligned I will always resize (round up) the host buffer (and period_bytes), so the host component will run with that scheduling, but rest of the pipeline will obey exactly the rate/sched. So for e.g. if you try to run pipeline 44 or 45 frames, host will push it up to next dma alignment (which probably in this case would be like running with 48 frames). Still rest of the pipeline will run with 44 as internally we don't have alignment restriction and host has ring buffer. As said I'm not sure how strictly the dma alignment should be obeyed...